app/controllers/api/usage_limits_controller.rb

```diff
@@ -59,11 +63,28 @@ def find_plan
     @plan = current_account.application_plans.find(params[:application_plan_id])
   end
 
+  def find_service
```

Comment:
where's this `@service` variable is being used?

Reply:
right, I understand that we are fetching the service for security, but when it comes to the instance variable... where do we use it? if it is only in the views, maybe it should be a helper, and if it is not used at all, there shouldn't be a variable saving it 😄 
same for most controllers here btw 😄

Reply:
The service is always needed in these controllers of the Admin Portal where the UI page is inside the "service section", i.e., when there's a service selected in the context switcher, the vertical nav bar expects a service for the links, etc. Moreover, finding the service is also the "trick" to make sure a 404 will be returned when the user does not have permission to read that specific service.